include/seqan/index/find2_index_approx.h

```diff
@@ -0,0 +1,593 @@
+            }
+            else
+            {
+                _schemeSearch(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + 1, s, blockIndex, TDir(), TDistanceTag());
+            }
+        }
+        _schemeSearchChildren(delegate, iter, needle, needleLeftIt, needleRightIt, errors, s, blockIndex, minErrorsLeftInBlock, TDir(), TDistanceTag());
+    }
+}
+
+template <typename TDelegate, typename TText, typename TIndex, typename TIndexSpec, typename TText2, size_t nbrBlocks, typename TDistanceTag>
+inline void _schemeSearch(TDelegate & delegate,
+                          Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > it,
+                          TText2 const & needle,
+                          Search<nbrBlocks> const & s,
+                          TDistanceTag /**/)
+{
+    _schemeSearch(delegate, it, needle, s.startPos, s.startPos + 1, 0, s, 0, Rev(), TDistanceTag());
+}
+
+template <typename TDelegate, typename TText, typename TIndex, typename TIndexSpec, typename TText2, size_t nbrBlocks, size_t N, typename TDistanceTag>
+inline void _schemeSearches(TDelegate & delegate,
+                          Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > it,
+                          TText2 const & needle,
+                          std::array<Search<nbrBlocks>, N> const & ss,
+                          TDistanceTag /**/)
+{
+    for (auto & s : ss)
+        _schemeSearch(delegate, it, needle, s, TDistanceTag());
+}
+
+// ----------------------------------------------------------------------------
+// Function _find()
+// ----------------------------------------------------------------------------
+
+template <size_t minErrors, size_t maxErrors, typename TText, typename TIndexSpec, typename TPattern, typename TDelegate, typename TDistanceTag>
+inline void
+_find(TDelegate & delegate,
+      Index<TText, BidirectionalIndex<TIndexSpec> > & index,
+      TPattern const & pattern,
+      TDistanceTag /**/)
+{
+    auto scheme = SearchSchemes<minErrors, maxErrors>::VALUE;
+    _schemeSearchComputeFixedBlocklength(scheme, length(pattern));
+    Iter<Index<TText, BidirectionalIndex<TIndexSpec> >, VSTree<TopDown<> > > it(index);
+    _schemeSearches(delegate, it, pattern, scheme, TDistanceTag());
+}
+
+template <size_t minErrors, size_t maxErrors, typename TText, typename TIndexSpec, typename TStringSetSpec, typename TPattern, typename TDelegate, typename TDistanceTag, typename TParallelTag>
+inline void
+_find(TDelegate & delegate,
+      Index<TText, BidirectionalIndex<TIndexSpec> > & index,
+      StringSet<TPattern, TStringSetSpec> const & patterns,
+      TDistanceTag /**/,
+      Tag<TParallelTag> /**/)
+{
+    SEQAN_OMP_PRAGMA(parallel for if (IsSameType<Tag<TParallelTag>, Parallel>::VALUE))
+    for (size_t seqNo = 0; seqNo < length(patterns); ++seqNo)
```

Comment:
Why not use: https://github.com/seqan/seqan/blob/028fd79c6e02b7255c95d390b965d922aa874bb6/include/seqan/parallel/parallel_algorithms.h#L232